message/messaging_service.cc

```diff
@@ -538,7 +539,6 @@ static constexpr unsigned do_get_rpc_client_idx(messaging_verb verb) {
     case messaging_verb::READ_DATA:
     case messaging_verb::READ_MUTATION_DATA:
     case messaging_verb::READ_DIGEST:
-    case messaging_verb::GOSSIP_DIGEST_ACK:
     case messaging_verb::DEFINITIONS_UPDATE:
```

Comment:
This was on purpose, to prevent a deadlock scenario. If A stuffs the A->B connection with SYN messages, and if B stuffs the B->A connection with SYN messages, the ACK messages will get stuck behind them.

I don't think there is risk in the patch (stuffing connections with SYN messages is unlikely), but also I don't understand how it solves #7986, or why d47033837ad6a6284bd0c60754a63f6a0413de74 caused a regression. My guess is that gossip does some heavy calculations and we need to reduce its shares to 100 or so.

Reply:
SYN message is no more than once per second, it could not stuff a connection. 

The d470338 makes gossip ack message runs inside statement group ( I do not understand how this could happen neither). Perhaps the contention introduced by ack message handler inside statement group makes latency high. Anyway, we should not put gossip handler inside statement group.

Reply:
I think the theoretical deadlock scenario does not justify putting GOSSIP_DIGEST_ACK inside the statement group. The ack message callback handles most of the gossip work.